T7-T9: Landing Page, Nav Bar, and Footer (new pr)

hferoz1 opened the pull request:
In this commit, I have:
-  changed everything from camelcase to kebab case
- removed unnecessary import statements 
- put the landing page, nav bar, and footer in separate js files
- created a better footer
- use Link from React Router to link the navbar options to the correct pages

hferoz1 committed:
T7-T9: Landing Page, Nav Bar, and Footer (new pr)

In this commit, I have:
-  changed everything from camelcase to kebab case
- removed unnecessary import statements 
- put the landing page, nav bar, and footer in separate js files
- created a better footer
- use Link from React Router to link the navbar options to the correct pages

hferoz1 commented:
A few screenshots of the updated UI!

![img1](https://github.com/user-attachments/assets/ec129b10-1d97-4418-9ece-5602f4b0a532)
![img2](https://github.com/user-attachments/assets/8348cc05-e8da-4234-896c-3eb16579e0ea)
![img3](https://github.com/user-attachments/assets/086daae4-30bc-4926-b881-ae02eded5b56)

kdata12 commented on frontend/package.json:
```diff
@@ -1,48 +1 @@
-  "author": "",
-  "license": "ISC"
-}
+{"name":"frontend","version":"0.1.0","private":true,"proxy":"http://localhost:8080","dependencies":{"@testing-library/jest-dom":"^5.17.0","@testing-library/react":"^13.4.0","@testing-library/user-event":"^13.5.0","axios":"^1.7.2","classnames":"^2.5.1","react":"^18.3.1","react-dom":"^18.3.1","react-router":"^6.23.1","react-router-dom":"^6.23.1","react-scripts":"5.0.1","router":"^1.3.8","web-vitals ...
```
Not sure what happened here, but can you reformat this to the it's original format?

kdata12 commented on frontend/src/App.js:
```diff
@@ -1,23 +1,30 @@
+import LandingPage from './components/LandingPage/LandingPage.js'; 
+import './components/LandingPage/LandingPage.module.css'; 
+
+export default function App() {
+  return (
+    <>
+      <NavBar />
+      <Routes>
+        <Route path="/" element={<LandingPage />} />
+        <Route path="/form" element={<FormPage />} />
+        <Route path="/sign-up" element={<Signup />} />
+        <Route path="/sign-in" element={<Signin />} />
+        <Route path="/reset-username-form" element={<ResetUsernameForm />} />
+        <Route path="/reset-password-form" element={<ResetPasswordForm />} />
+      </Routes>
+      <Footer />
+    </>
+  );
+}
```
nice!

kdata12 commented on frontend/src/components/LandingPage/NavBar.js:
```diff
@@ -0,0 +1,20 @@
+function NavBar() {
+  return (
+    <header>
+      <div className={classNames(styles['nav-bar'])}>
```
minor adjustment, but we can use the `<nav> </nav>` tag here just to be more semantic instead of a div.

kdata12 commented on frontend/src/components/authorization/SignIn.js:
```diff
@@ -1,59 +1,59 @@
+  const [userEmail, setUserEmail] = useState (''); 
+  const [userPassword, setUserPassword] = useState ('');
+
+  const handleSignInSubmit = async (_event) => { 
+    const response = await axios.post('/sign-in', {
+      email: userEmail,
+      password: userPassword,
+    }); 
+    console.alert("sign in was successful for ", response.data)
+  }
```
How are we handling errors from the server here? I suggest using a try and catch block here to gracefully handle errors.

kdata12 replied:
I'm also not sure if this is your code or your teammate's code.

hferoz1 replied:
this is Jonathan's code

kdata12 commented on frontend/src/components/authorization/SignUp.js:
```diff
@@ -1,68 +1,68 @@
+import axios from 'axios';
+import {useState} from 'react';
+
+function SignUp() {
```
is this your code too?

hferoz1 replied:
this is jonathon's code too

kdata12 commented on frontend/src/form.module.css:
```diff
@@ -1,79 +1,79 @@
+    box-shadow: 2px 2px 5px gray;
+}
+
+form {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    gap: 30px;
+    width: 40vh;
+    flex: 0;
+}
```
I would advise against using a css element selector here because it will affect all form elements in other files, not just your own file. This css will apply to all future form elements, so be careful using this.

hferoz1 replied:
this is not part of my code, i think this is jonathon's

kdata12 commented on frontend/src/form.module.css:
```diff
@@ -1,79 +1,79 @@
+    padding-left: 10px;
+}
+
+h1 {
+    font-size: 30px;
+}
```
Just like the form element selector, i would advise against using an element selector.

hferoz1 replied:
jonathon's code too i think

kdata12 commented on frontend/src/components/LandingPage/LandingPage.module.css:
```diff
@@ -0,0 +1,204 @@
+  background-color: #f0f0f0;
+}
+
+h1 {
+  text-align: center;
+  color: #38023b;
+  padding-top: 20px;
+}
```
This h1 style will be applied to all h1 tags in your codebase, so if you want to locally scope this style, use a classname selector instead of an h1 tag selector

kdata12 commented on frontend/src/components/LandingPage/LandingPage.module.css:
```diff
@@ -0,0 +1,204 @@
+  color: #f5f3f5;
+}
+
+img {
+  max-width: 100%;
+  max-height: 190px;
+  border: 2px solid #38023b;
+  border-radius: 20px;
+}
```
same here, use a class selector instead of an element selector

kdata12 reviewed (commented):
Good work hafsa. The landing page, nav and footer are looking good! Just left a few comments. It also seemed like you created changes to some files that weren't yours so I got confused and left some comments on those unrelated files, you can ignore them.

hferoz1 committed:
Add files via upload

In this commit, I have:
- switched div tag to nav tag in NavBar.js
- updated element selectors to class selectors

No updates to UI layout!

kdata12 reviewed (approved):
lgtm!
